### PR TITLE
Check if attachment with type supplied has already been filed for a case

### PR DESCRIPTION
**Pull request by rsachdevch**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,10 +4,11 @@ go 1.12
 
 require (
 	github.com/companieshouse/api-sdk-go v0.1.13
-	github.com/companieshouse/chs.go v1.2.2
+	github.com/companieshouse/chs.go v1.2.4
 	github.com/companieshouse/go-sdk-manager v0.1.7
 	github.com/companieshouse/go-session-handler v0.1.5
 	github.com/companieshouse/gofigure v0.1.4
+	github.com/companieshouse/lfp-pay-api v0.0.0-20210412134412-60173480c86c
```
Comment (emead) [inline go.mod:11]: Need to run `go mod tidy` to remove this

File: handlers/attachment_resource.go [hunk 2]
```diff
@@ -38,7 +38,13 @@ func HandleSubmitAttachment(svc dao.Service) http.Handler {
 		}
 
 		// Validate that the provided attachment details are correct
-		validationErrs := service.ValidateAttachmentDetails(attachmentType, header)
+		validationErrs, err := service.ValidateAttachmentDetails(svc, transactionID, attachmentType, header)
+		if err != nil {
+			log.ErrorR(req, fmt.Errorf("error retrieving attachments from db: [%s]", err))
```
Comment (emead) [inline handlers/attachment_resource.go:43]: Might not be a DB error
```suggestion
			log.ErrorR(req, fmt.Errorf("error validating attachment details: [%s]", err))
```

File: handlers/attachment_resource_test.go [hunk 3]
```diff
@@ -12,15 +12,17 @@ import (
 	"os"
 	"testing"
 
+	"github.com/jarcoal/httpmock"
```
Comment (emead) [inline handlers/attachment_resource_test.go:15]: Group imports together

File: handlers/attachment_resource_test.go [hunk 4]
```diff
@@ -12,15 +12,17 @@ import (
...
+
 	"github.com/companieshouse/chs.go/log"
 	"github.com/companieshouse/go-session-handler/httpsession"
 	"github.com/companieshouse/go-session-handler/session"
 	"github.com/companieshouse/insolvency-api/dao"
+	"github.com/companieshouse/insolvency-api/mocks"
```
Comment (emead) [inline handlers/attachment_resource_test.go:21]: Duplicated import

File: dao/mongo.go [hunk 5]
```diff
@@ -377,3 +377,38 @@ func (m *MongoService) AddAttachmentToInsolvencyResource(transactionID string, f
...
+
+	err = storedAttachments.Decode(&insolvencyResource)
+	if err != nil {
+		log.Error(err)
+		return nil, err
+	}
+
+	// Return an empty array instead of nil to distinguish from insolvency case
+	// not found
+	if insolvencyResource.Data.Attachments == nil {
```
Comment (emead) [inline dao/mongo.go:409]: Also check if Data is nil, to avoid null pointer error

Comment (Joseph-Nash) [reply dao/mongo.go:409]: Decided this is covered by other checks, data is unlikely to be empty. Marking as resolved.

**Review by emead — APPROVED** [review]

**Pull request merged**